src/lsp-client.ts

```diff
@@ -18,7 +26,47 @@ export interface LspClient {
 }
 
 export class LspClientImpl implements LspClient {
-    constructor(protected connection: lsp.Connection) {
+    private clientCapabilities?: lsp.ClientCapabilities;
+
+    constructor(protected connection: lsp.Connection) {}
+
+    setClientCapabilites(capabilites: lsp.ClientCapabilities): void {
+        this.clientCapabilities = capabilites;
+    }
+
+    createProgressReporter(): ProgressReporter {
+        let workDoneProgress: Promise<lsp.WorkDoneProgressServerReporter> | undefined;
+        return {
+            begin: (message = '') => {
+                if (this.clientCapabilities?.window?.workDoneProgress) {
+                    workDoneProgress = this.connection.window.createWorkDoneProgress();
+                    workDoneProgress
+                        .then((progress) => {
+                            progress.begin(message);
+                        })
+                        .catch(() => {});
```

Comment:
can the `createWorkDoneProgress` throw an exception?
is it ok to silence the exception `catch(() => {});` ?

Reply:
If it doesn't throw an exception, 
IMO the async/await syntax is more easier to read. 

```ts
            begin: async (message = '') => {
                if (this.clientCapabilities?.window?.workDoneProgress) {
                    const progress = await this.connection.window.createWorkDoneProgress();
                    progress.begin(message);
                }
            },
```

Reply:
It technically can reject since the client can refuse to create the token.

This code was inspired by something I found in pyright but personally I think it makes more sense this way.

We don't really need to await those methods but when using `async` function it is sort of expected that those are awaited and I don't want to do that.

Reply:
Github has a bug on my end... 
I do not see the resolve conversation button again :)

Reply:
Happens to me to sometimes and I can't figure out what's the Github's logic there...